### PR TITLE
Backported getLocalTime from ESP32 Arduino

### PR DESCRIPTION
**Pull request by h1aji**
Backporting getLocalTime from ESP32 for better compatibility.
Also fixed configTzTime

### CODE REVIEW

File: cores/esp8266/Arduino.h [hunk 1]
```diff
@@ -269,16 +269,15 @@ void setTZ(const char* tz);
 void configTime(int timezone, int daylightOffset_sec, const char* server1,
     const char* server2 = nullptr, const char* server3 = nullptr);
 
-void configTime(const char* tz, const char* server1,
```
Comment (earlephilhower) [inline cores/esp8266/Arduino.h:272]: Why would you delete the alternative API?  Restore, please.

File: cores/esp8266/time.cpp [hunk 2]
```diff
@@ -187,26 +187,44 @@ void configTime(int timezone_sec, int daylightOffset_sec, const char* server1, c
...
+void setTZ(const char* tz)
+{
     char tzram[strlen_P(tz) + 1];
     memcpy_P(tzram, tz, sizeof(tzram));
     setenv("TZ", tzram, 1/*overwrite*/);
     tzset();
 }
 
-void configTime(const char* tz, const char* server1, const char* server2, const char* server3)
+void configTzTime(const char* tz, const char* server1, const char* server2, const char* server3)
```
Comment (earlephilhower) [inline cores/esp8266/time.cpp:198]: Don't rename a well-known function.  That breaks backwards compatibility for no reason.

File: cores/esp8266/time.cpp [hunk 3]
```diff
@@ -187,26 +187,44 @@ void configTime(int timezone_sec, int daylightOffset_sec, const char* server1, c
...
 }
 
+/*
+ * Backported from ESP32 Arduino
+ */
+bool getLocalTime(struct tm * info, uint32_t ms)
+{
+    uint32_t start = millis();
+    time_t now;
+    while((millis()-start) <= ms)
```
Comment (earlephilhower) [inline cores/esp8266/time.cpp:215]: Indent has gone wacky in this function.

**Review by earlephilhower — CHANGES_REQUESTED** [review]

File: cores/esp8266/time.cpp (hunk 2 above)
Comment (d-a-v) [reply cores/esp8266/time.cpp]: FYI, it is already [there](https://github.com/esp8266/Arduino/blob/55ef3e73978ad010d44ad7e4c9bd4b4f32455989/cores/esp8266/Arduino.h#L276-L280)

**Pull request closed without merging**